_posts/2021-04-15-jesse-frost.md

```diff
@@ -0,0 +1,81 @@
+authenticate its operations: they are the keys to freedom in the digital age.
+However, the requirements for Bitcoin key management are becoming more demanding
+with innovative off-chain protocols such as the [Lightning Network][ln] and
+[Discreet Log Contracts][dlc] that introduce new roles and responsibilities for
+keys. In addition, Bitcoin's relentless rise in value creates a need for
+sophisticated systems that can help make rubber-hose cryptanalysis uneconomical.
+
+Off-chain protocols facilitate smart contracts that are highly scalable and
+private, which enable a vast array of new markets and financial instruments.
+However, many of these protocols present challenges from a key management
+perspective because they rely on “hot” keys that must be readily accessible and
+frequently reused, unlike safer "cold" keys that are rarely accessed and never
+reused. With the introduction of [Taproot][bip341], we can begin to bridge this
+divide between a "hot" and "cold" key with distributed key generation and
+threshold signing.
+
+Distributed key generation uses secure multi-party computation ([MPC][mpc wiki])
+to generate a pair of public and private keys with multiple parties such that no
+party learns the private key. Instead, each party learns a "share" of the
+private key, and when a threshold of these parties cooperate they can use their
+shares to generate signatures for the key. The threshold is configured by the
+parties when the key is generated. For example, if 5 parties generated a key
+with a threshold of 3, then at least 3 of the parties would need to cooperate to
+produce a signature (i.e. 3-of-5).
+
+This means that for the entire lifecycle of the private key–from generation to
+expiration–it is never constructed or represented on any device in any form
+(assuming a threshold of shares has not been compromised). With Taproot, we can
+leverage the linearity of Schnorr signatures to perform signing with 3
+communication rounds (or 2 rounds with limited concurrency), unlike the ECDSA
+variants that are complex and require many more rounds, and, with the recent
+introduction of Flexible Round-Optimized Schnorr Threshold Signatures
+([FROST][frost whitepaper]), signing can now be done in a single round with
+unlimited concurrency.
+
+FROST is a relatively new scheme yet some notable open-source implementations
+have already been developed, such as [isislovecruft/frost-dalek][frost dalek github],
+a Rust implementation using the Ristretto group,
+[taurusgroup/frost-ed25519][frost ed25519 github], a Go
+implementation using Ed25519, and [ZcashFoundation/redjubjub][redjubjub github], a Rust
+implementation using Jubjub. However, none of these implementations is currently
+compatible with Bitcoin, which uses secp256k1 and, if/when Taproot activates,
+BIP-340 Schnorr signatures. My plan is to implement a Bitcoin compatible FROST
+in C built on [ElementsProject/secp256k1-zkp][secp256k1-zkp github].
```

Comment:
"and, if/when Taproot activates, BIP-340 Schnorr signatures" is a bit difficult to parse. Perhaps just remove it. BIP 340 Schnorr signatures also use secp256k1.